core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -330,18 +330,45 @@ object SparkSubmit extends CommandLineUtils {
     args.archives = Option(args.archives).map(resolveGlobPaths(_, hadoopConf)).orNull
 
     // In client mode, download remote files.
+    var localPrimaryResource: String = null
+    var localJars: String = null
+    var localPyFiles: String = null
     if (deployMode == CLIENT) {
-      args.primaryResource = Option(args.primaryResource).map {
+      localPrimaryResource = Option(args.primaryResource).map {
         downloadFile(_, targetDir, args.sparkProperties, hadoopConf)
       }.orNull
-      args.jars = Option(args.jars).map {
+      localJars = Option(args.jars).map {
         downloadFileList(_, targetDir, args.sparkProperties, hadoopConf)
       }.orNull
-      args.pyFiles = Option(args.pyFiles).map {
+      localPyFiles = Option(args.pyFiles).map {
         downloadFileList(_, targetDir, args.sparkProperties, hadoopConf)
       }.orNull
     }
 
+    if (clusterManager == YARN) {
+      def isNoneFsFileExist(paths: String): Boolean = {
+        Option(paths).exists { p =>
+          p.split(",").map(_.trim).filter(_.nonEmpty).exists { path =>
+            val url = Utils.resolveURI(path)
+            url.getScheme match {
+              case "http" | "https" | "ftp" => true
+              case _ => false
+            }
+          }
+        }
+      }
+
+      // Spark on YARN doesn't support upload remote resources from http, https or ftp server
+      // directly to distributed cache, so print a warning and exit the process.
+      if (isNoneFsFileExist(args.jars) ||
```

Comment:
In the yarn mode, directly uploading remote resources from http(s) to hadoop FS will throw exception, so here guard out such scenario. 

We can also support this by filtering and downloading http(s) resources to local and re-upload to Hadoop FS. But it makes the code here quite complicated, so I chose to simply guard out such scenario.